### PR TITLE
[#3] [UI] As a user, I can see the Splash screen

### PR DESCRIPTION
**Pull request by hoangnguyen92dn**
Close https://github.com/hoangnguyen92dn/survey-flutter-ic/issues/3

## What happened 👀

Add Splash screen as the default screen will be shown when launch application

## Insight 📝

- Add `flutter_svg` library to support `.svg` format and `flutter_gen` to manage resources
- Apply `SystemOverlayStyle` to transparent the status bar

```
SystemChrome.setSystemUIOverlayStyle(
        const SystemUiOverlayStyle(statusBarColor: Colors.transparent));
```

- Upgrade `flutter_gen_runner` version to resolve incompatible version: https://github.com/FlutterGen/flutter_gen/issues/352
- Notice: I attempted to use the `bg_splash` as svg but the rendering shows weird image so I have to go with `.png` format.

## Proof Of Work 📹

https://user-images.githubusercontent.com/6950766/223949802-5d0e9ee3-b5bc-4f87-94df-ad3dc6b60e4b.mp4

### CODE REVIEW

**Review by doannimble — COMMENTED** [review]
rest lgtm

File: lib/ui/splash_screen.dart [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Container(
+        decoration: BoxDecoration(
+          image: DecorationImage(
+              image: AssetImage(Assets.images.bgSplash.path), fit: BoxFit.fill),
+        ),
+        child: Center(
+          child: AnimatedOpacity(
+            duration: const Duration(milliseconds: 1000),
```
[the remark this answers is not in the document]
Comment (hoangnguyen92dn) [reply lib/ui/splash_screen.dart:50]: 👌 Let me extract to constant

File: lib/main.dart [hunk 2]
```diff
@@ -21,15 +22,13 @@ class MyApp extends StatelessWidget {
     routes: <GoRoute>[
       GoRoute(
         path: routePathRootScreen,
+        builder: (BuildContext context, GoRouterState state) =>
```
[the remark this answers is not in the document]
Comment (hoangnguyen92dn) [reply lib/main.dart:25]: Yup. We will refactor the navigator then so let do it later.

**Review by doannimble — APPROVED** [review]
LGTM 🚀

**Review by Wadeewee — APPROVED** [review]
LGTM 🚀

**Pull request merged**